### PR TITLE
hlint: add manpage

### PR DESCRIPTION
**Pull request by teto**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/haskell-modules/configuration-common.nix [hunk 1]
```diff
@@ -88,6 +88,13 @@ self: super: {
     };
   }) super.git-annex;
 
+  hlint = super.hlint.overrideAttrs(oa: {
+    postInstall = ''
+      mkdir -p $out/share/man
+      cp data/hlint.1 $out/share/man
+      '';
+  });
```
Comment (sternenseemann) [inline pkgs/development/haskell-modules/configuration-common.nix:96]: * `overrideCabal` makes sure modifications are not later rolled back by another chained haskell override (we want to override the args to `haskellPackages.mkDerivation`, not `stdenv.mkDerivation`).
* `drv` is used  generally as “old args” variable in the `configuration*` files, so stick with that.
* Make sure to concatenate the `postInstall` with the old one (if any).

```suggestion
  hlint = overrideCabal (drv: {
    postInstall = ''
      mkdir -p $out/share/man
      cp data/hlint.1 $out/share/man
    '' + drv.postInstall or "";
  }) super.hlint;
```

**Review by sternenseemann — CHANGES_REQUESTED** [review]
Please move this override into `configuration-nix.nix` as well, since this is not a temporary workaround type override.

File: pkgs/development/haskell-modules/configuration-common.nix (hunk 1 above)
Comment (sternenseemann) [reply pkgs/development/haskell-modules/configuration-common.nix:96]: If you feel like it you could also use `install -Dm644 data/hlint.1 -t "$out/share/man"` to do directory creation and copying in one step.

**Comment by teto** [comment]
thanks for the onpoint review. There is 0 line left from my initial PR but it is much better now ;)

File: pkgs/development/haskell-modules/configuration-nix.nix [hunk 2]
```diff
@@ -818,6 +818,12 @@ self: super: builtins.intersectAttrs super {
     platforms = pkgs.lib.platforms.x86;
   } super.geomancy;
 
+  hlint = overrideCabal (drv: {
+    postInstall = ''
+      install -Dm644 data/hlint.1 -t "$out/share/man"
```
Comment (sternenseemann) [inline pkgs/development/haskell-modules/configuration-nix.nix:823]: Oh, also completely missed this: needs to go into the `man1` subdirectory or man page viewers won't find it:

```suggestion
      install -Dm644 data/hlint.1 -t "$out/share/man/man1"
```

**Review by sternenseemann — CHANGES_REQUESTED** [review]

**Pull request merged**